tincr/io/design/tincr_checkpoints.tcl

```diff
@@ -178,6 +181,24 @@ proc ::tincr::read_tcp {args} {
             ::tincr::print_verbose "Done routing...($diff_time seconds)"
         }
     }
+	
+    # Complete the route for nets with a hierarchical source port.
+    # The same warning/bug described above occurs when trying to specify the ROUTE string of a net
+    # that is a hierarchical port (placed or unplaced port with no driver).
+    # Work around is also to have Vivado route these nets for us.
+    if {$link_mode=="out_of_context"} {
+	set diff_time 0
```

Comment:
Why is `diff_time` declared in this scope?

Reply:
It is actually first declared on line 170. I am resetting it back to 0 on this line in case there are hierarchical ports that need routing.